### PR TITLE
Do some cleanups for hashmaps

### PR DESCRIPTION
**Pull request by Manishearth**
@mystor noticed some things whilst reading through the hashmap RawTable code.

Firstly, in RawTable we deal with this hash_offset value that is the offset of the list of hashes from the buffer start. This is always zero, and this isn't consistently used (which means that we would have bugs if we set it to something else). We should just remove this since it doesn't help us at all.

Secondly, the probing length tag is not copied when cloning a raw table. This is minor and basically means we do a bit more work than we need on further inserts on a cloned hashmap.

r? @Gankro

### CODE REVIEW

**Comment by bluss** [comment]
@arthurprs is probably interested in this PR

File: src/libstd/collections/hash/table.rs [hunk 1]
```diff
@@ -768,7 +767,7 @@ impl<K, V> RawTable<K, V> {
         // This is great in theory, but in practice getting the alignment
         // right is a little subtle. Therefore, calculating offsets has been
         // factored out into a different function.
-        let (alignment, hash_offset, size, oflo) = calculate_allocation(hashes_size,
+        let (alignment, size, oflo) = calculate_allocation(hashes_size,
                                                                         align_of::<HashUint>(),
```
Comment (arthurprs) [inline src/libstd/collections/hash/table.rs]: Nit: lines bellow need some formating adjustments.

File: src/libstd/collections/hash/table.rs [hunk 2]
```diff
@@ -1157,6 +1156,7 @@ impl<K: Clone, V: Clone> Clone for RawTable<K, V> {
             }
 
             new_ht.size = self.size();
+            new_ht.set_tag(self.tag());
```
Comment (arthurprs) [inline src/libstd/collections/hash/table.rs]: Good point, I overlooked that before.

**Review by arthurprs — APPROVED** [review]
Nice!

**Comment by Manishearth** [comment]
Fixed

**Comment by bluss** [comment]
@bors r+ rollup

Thanks @Manishearth and @arthurprs

**Pull request merged**